File: 코딩 기초 트레이닝/리스트 자르기-p181897/리스트 자르기.js
// https://school.programmers.co.kr/learn/courses/30/lessons/181897?language=javascript

function solution(n, slicer, num_list) {
    if(n === 1) {
        return num_list.slice(0, slicer[1]+1);
    } else if( n === 2) {
        return num_list.slice(slicer[0], );
    } else if( n === 3) {
        return num_list.slice(slicer[0], slicer[1]+1);
    } else {
        return num_list.filter((n, idx) => {
            if (idx % slicer[2] === slicer[0] % slicer[2] && idx >= slicer[0] && idx <= slicer[1]) return n;
        })
    }
}

File: 코딩 기초 트레이닝/리스트 자르기-p181897/리스트 자르기.py
# https://school.programmers.co.kr/learn/courses/30/lessons/181897

def solution(n, slicer, num_list):
    # n == 1 0 ~ b
    # n == 2 a ~
    # n == 3 a ~ b
    # n == 4 a ~ b c 간격

    if n == 1:
        b = slicer[1]
        return num_list[0: b+1]
    elif n == 2:
        a = slicer[0]
        return num_list[a:]
    elif n == 3:
        a = slicer[0]
        b = slicer[1]
        return num_list[a: b+1]
    else:
        a = slicer[0]
        b = slicer[1]
        c = slicer[2]
        return num_list[a:b+1:c]
